### DISCUSSION TITLE
Shadowsocks username

### DISCUSSION THREAD

**Discussion by seb3point0** [qdm12/gluetun#509] [Q&A] [answered]
What's does the Shadowsocks username default to?

**Comment by qdm12** [comment, marked as answer]
There is no username only a password for Shadowsocks. Where did you see there was a username?

**Reply by seb3point0** [reply]
I was under the impression that Shadowsocks was a standard socks5 server and that one could connect to it using this scheme `curl --socks5 user:pass@localhost:8388 https://ifconfig.io`.

**Reply by qdm12** [reply]
No it's not. It's based on socks5 with an encryption layer, so it's not respecting the socks5 spec really. You have to use a Shadowsocks client (https://shadowsocks.org/en/download/clients.html) to connect to it, they have clients for pretty much every platform.

**Reply by seb3point0** [reply]
I realized this after reading up about it. I'm using the proxy with Puppeteer so I'll probably stick to HTTP for now, for simplicity's sake. Thanks!

**Reply by qdm12** [reply]
Cool! You might want to subscribe to https://github.com/qdm12/gluetun/issues/278 I'll try to do it soon, such that your HTTP proxy connection will be encrypted with TLS.
